testing/trino-product-tests-launcher/bin/ignoredModuleCalculator.py

```diff
@@ -0,0 +1,98 @@
+#!/usr/bin/env python3
+
+import argparse
+import collections
+import csv
+import logging
+import os
+import sys
+import xml.etree.ElementTree as xml
+
+JAVA_PACKAGE_SEPARATOR = '.'
+PATH_TO_PACKAGE_TRANSLATION = str.maketrans(os.sep, JAVA_PACKAGE_SEPARATOR)
+COVERAGE_TO_CLASSFILE_TRANSLATION = str.maketrans(JAVA_PACKAGE_SEPARATOR, '$')
+
+
+def main():
+    parser = argparse.ArgumentParser(
+        description="List ignored modules for a PT suite based on coverage and Trino root POM."
+    )
+    parser.add_argument(
+        "-c",
+        "--coverage",
+        type=argparse.FileType("r"),
+        help="A Jacoco CSV file with coverage",
+    )
+    parser.add_argument(
+        "-p",
+        "--pom",
+        type=argparse.FileType("r"),
+        help="Trino root POM file",
+    )
+    parser.add_argument(
+        "-o",
+        "--output",
+        type=argparse.FileType("w"),
+        default=sys.stdout,
+        help="Filename to write output to",
+    )
+    parser.add_argument(
+        "-d",
+        "--debug",
+        action="store_const",
+        dest="loglevel",
+        const=logging.DEBUG,
+        default=logging.WARNING,
+        help="Print lots of debugging statements",
+    )
+
+    args = parser.parse_args()
+    logging.basicConfig(level=args.loglevel)
+    build(args.coverage, args.pom, args.output)
+
+
+def build(coverage_file, pom_file, output_file):
+    coverage_reader = csv.DictReader(coverage_file)
+    coverage = collections.defaultdict(lambda: collections.defaultdict(int))
+    for row in coverage_reader:
+        logging.debug("row: %s", row)
+        class_name = row["CLASS"].split("new ")[0].translate(COVERAGE_TO_CLASSFILE_TRANSLATION)
+        coverage[row["PACKAGE"]][class_name] += int(row["INSTRUCTION_COVERED"])
+    logging.debug("coverage: %s", coverage)
+
+    root_path = os.path.dirname(pom_file.name)
+    pom = xml.parse(pom_file)
+    ns = {'xmlns': pom.getroot().tag.split("{")[1].split("}")[0]}
+    for element in pom.findall('xmlns:modules/xmlns:module', namespaces=ns):
+        module = element.text
+        module_root = os.path.join(root_path, module)
+        logging.debug("module_root: %s", module_root)
+        module_classes_root = os.path.join(module_root, "target", "classes")
+        module_test_root = os.path.join(module_root, "target", "test-classes")
+        if unused(module_classes_root, coverage) and unused(module_test_root, coverage):
+            output_file.write("%s\n" % module)
+
+
+def unused(source_root, coverage):
+    for root, dirs, files in os.walk(source_root):
+        logging.debug("root: %s, dirs: %s, files: %s", root, dirs, files)
+        for file in files:
+            if file.endswith(".class"):
```

Comment:
Nit: if you reverse this condition and do a `continue` you can avoid one level of indentation. It'll make the whole method much more readable.